exonum/src/events/noise/wrappers/sodium_wrapper/wrapper.rs

```diff
@@ -140,12 +141,13 @@ impl NoiseWrapper {
         buf.extend_from_slice(&msg_len_buf);
 
         debug_assert_eq!(len, encoded_message.len());
```

Comment:
Maybe off-topic, but I think should be fixed here: this assert is useless because of `let encoded_message = &encrypted_message[0..len]` above (which is useless on its own right, too). The assertion should be performed immediately after forming the message. You can include writing the message length into the message initialization, too:

```rust
let len = encrypted_msg_len(msg.len());
let mut encrypted_message = vec![0; len + HEADER_LENGTH];
// It could be worth trying to research vectors with uninitialized values, or `BytesMut`.

LittleEndian::write_u32(&mut encrypted_message[..HEADER_LENGTH], len as u32);

const CHUNK_LEN: usize = MAX_MESSAGE_LENGTH - TAG_LENGTH;
for (i, msg) in msg.chunks(CHUNK_LEN).enumerate() {
    // ideally, it could be worth to avoid reallocations here
    // by making `self.write` accept `&mut [u8]`.
    let written = self.write(msg)?;

    let start = i * CHUNK_LEN;
    encrypted_message[start..(start + CHUNK_LEN)].copy_from_slice(&written);
}
```